src/js/core/directives/ui-grid-column-menu.js

```diff
@@ -401,8 +401,11 @@ function ($timeout, gridUtil, uiGridConstants, uiGridColumnMenuService, $documen
           $scope.hideMenu( true );
 
           if ($scope.col) {
-            //Focus on the menu button
-            gridUtil.focus.bySelector($document, '.ui-grid-header-cell.' + $scope.col.getColClass()+ ' .ui-grid-column-menu-button', $scope.col.grid, false);
+            //Focus on the menu button if this is shown, otherwise focus the column header
+            var elementToFocus = '.ui-grid-header-cell.' + $scope.col.getColClass();
+            elementToFocus += ($scope.grid.options.showColumnMenuButtons && $scope.col.colDef.showColumnMenuButton !== false)? 
```

Comment:
This is redundant. if $scope.grid.options.showColumnMenuButton returns truthy, then it is obviously different than false. You only need one of these checks.

Reply:
They are not the same. Please look carefully, the first is general and the second is column specific.